src/platform/silabs/SilabsConfig.cpp

```diff
@@ -65,16 +71,9 @@ namespace Internal {
 
 CHIP_ERROR SilabsConfig::Init()
 {
-#ifndef BRD4325A // TODO: fix semaphore usage in nvm3_lock for siwx917. use weak implementation for that board instead
-    nvm3_Sem = xSemaphoreCreateBinaryStatic(&nvm3_SemStruct);
-
-    if (nvm3_Sem == NULL)
-    {
-        return CHIP_ERROR_NO_MEMORY;
-    }
-    xSemaphoreGive(nvm3_Sem);
-#endif // not BRD4325A
-    return MapNvm3Error(nvm3_open(nvm3_defaultHandle, nvm3_defaultInit));
+    // nvm3_Sem is created in nvm3_lockBegin()
+    // nvm3_open() is  called by nvm3_initDefault() in sl_platform_init() in SLC-generated sl_event_handler.c
```

Comment:
The nvm3_open() can be called more than once without risks. Maybe we should keep it, even if it is a duplicated call in our case, for customers that would use SilabsConfig:: storage APIs without using slc or our pregenerated boards, like a custom board init.

Reply:
Restored the call